include/rocksdb/env.h

```diff
@@ -473,16 +470,6 @@ class RandomAccessFile {
   virtual Status Read(uint64_t offset, size_t n, Slice* result,
                       char* scratch) const = 0;
 
-  // Used by the file_reader_writer to decide if the ReadAhead wrapper
-  // should simply forward the call and do not enact buffering or locking.
-  virtual bool ShouldForwardRawRequest() const {
```

Comment:
Two options here. We could either:
1. Remove these APIs (ShouldForwardRawRequest and EnableReadAhead), but take on the additional responsibility of making changes in the user code by using either version guards or deleting the reference entirely. (the user code around these seems to be just a thin wrapper, afaict). 
2. Leave these public APIs here, as deprecated, so that users are not affected. 

I personally prefer option 1 so that the code is cleaner everywhere. But I'll leave the final call to @siying .